src/main/java/com/elixr/poc/exception/GlobalExceptionHandler.java

```diff
@@ -17,17 +19,44 @@ public class GlobalExceptionHandler {
 
     /**
      * Handles MethodArgumentNotValidException
+     *
      * @param methodArgumentNotValidException
      * @return
      */
     @ExceptionHandler(MethodArgumentNotValidException.class)
     public ResponseEntity<?> handleMethodArgumentNotValidException(MethodArgumentNotValidException methodArgumentNotValidException) {
         List<String> errorList = new ArrayList<>();
         for (final FieldError error : methodArgumentNotValidException.getBindingResult().getFieldErrors()) {
-            errorList.add(error.getDefaultMessage());
+            errorList.add(error.getField()+" "+ MessagesUtil.getMessage(MessagesKeyEnum.ENTITY_MANDATORY_FIELD_MISSING.getKey()));
         }
-       PostErrorResponse errorResponse = PostErrorResponse.builder().errorMessage(errorList)
-                .build();
+        PostErrorResponse errorResponse = PostErrorResponse.builder().errorMessage(errorList).build();
         return new ResponseEntity<>(errorResponse, HttpStatus.BAD_REQUEST);
     }
+
+    /**
+     * URL exceptions are handled.
+     *
+     * @param globalException
+     * @return
+     */
+    @ExceptionHandler(IdNotFoundException.class)
+    public ResponseEntity handleGlobalException(IdNotFoundException globalException) {
+        CommonResponse commonResponse = CommonResponse.builder().success(false)
+                .errorMessage(MessagesUtil.getMessage(MessagesKeyEnum.ENTITY_ID_DOES_NOT_EXISTS.getKey())).build();
+        return new ResponseEntity<>(commonResponse, HttpStatus.NOT_FOUND);
+    }
+
+    /**
+     * Generic exceptions are handled.
+     *
+     * @param exception
+     * @return
+     */
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity handleGenericException(Exception exception) {
+        CommonResponse commonResponse = CommonResponse.builder().success(false)
+                .errorMessage(exception.getLocalizedMessage()).build();
+        return new ResponseEntity<>(commonResponse, HttpStatus.BAD_REQUEST);
```

Comment:
Can you try to create an internal server error scenario and check what is the http response code getting returned? Please post the response you get here.